### PR TITLE
Improve type hints in UniqueGenerator

### PR DESCRIPTION
**Pull request by spawnia**
### What is the reason for this PR?

- [x] Fixed an issue

With version 1.15.0 of Faker, PHPStan started showing static analysis issues related to the usage of `unique()`.

### Author's checklist

- [x] Follow the [Contribution Guide](https://github.com/FakerPHP/Faker/blob/main/.github/CONTRIBUTING.md)
- ~[ ] New features and changes are [documented](https://github.com/FakerPHP/fakerphp.github.io)~

### Summary of changes

- Clarifies the intent of the code
- Aids autocompletion
- Aids static analysis tools

### Review checklist

- [ ] All checks have passed
- [ ] Changes are approved by maintainer

### CODE REVIEW

**Comment by pimjansen** [comment]
Valid but it is a breaking change by typing the constructor

**Comment by jeremyFreeAgent** [comment]
Thanks!

File: src/Faker/UniqueGenerator.php [hunk 1]
```diff
@@ -3,13 +3,27 @@
...
+     */
     protected $generator;
+
+    /**
+     * @var int
+     */
     protected $maxRetries;
+
+    /**
+     * @var array<string, array<string, null>>
```
Comment (pimjansen) [inline src/Faker/UniqueGenerator.php:25]: This is not valid i assume? In fact there could be dates / int / string returned and so on if im correct right?

Comment (spawnia) [reply src/Faker/UniqueGenerator.php:32]: I left an explanatory comment for you.

Comment (pimjansen) [reply src/Faker/UniqueGenerator.php:32]: But the array can hold more then just an array with strings only if im correct

Comment (spawnia) [reply src/Faker/UniqueGenerator.php:32]:
```
$this->uniques[$name] = [];
$this->uniques[$name][serialize($res)] = null;
```

These are the only two writes to the array.

**Review by pimjansen — APPROVED** [review]

**Comment by pimjansen** [comment]
Thanks!

**Pull request merged**